flow/record/adapter/gcs.py

```diff
@@ -0,0 +1,87 @@
+import logging
+import re
+from fnmatch import fnmatch
+from typing import Iterator, Union
+
+from google.cloud.storage.client import Client
+from google.cloud.storage.fileio import BlobReader, BlobWriter
+
+from flow.record.adapter import AbstractReader, AbstractWriter
+from flow.record.base import Record, RecordAdapter
+from flow.record.selector import CompiledSelector, Selector
+
+__usage__ = """
+Google Cloud Storage adapter
+---
+Read usage: rdump gcs://[BUCKET_ID]/path?project_id=[PROJECT_ID]
+[PROJECT_ID]: Google Cloud Project ID
+[BUCKET_ID]: Bucket ID
+[path]: Path to look for files, with support for glob-pattern matching
+
+Write usage: rdump gcs://[BUCKET_ID]/path?project_id=[PROJECT_ID]
+[PROJECT_ID]: Google Cloud Project ID
+[BUCKET_ID]: Bucket ID
+[path]: Path to write records to
+"""
```

Comment:
Made the usage string a bit more concise, and also changed `project_id` -> `project` and make it optional.

```suggestion
__usage__ = """
Google Cloud Storage adapter
---
Read usage: rdump gcs://[BUCKET_ID]/path?project=[PROJECT]
Write usage: rdump -w gcs://[BUCKET_ID]/path?project=[PROJECT]

[BUCKET_ID]: Bucket ID
[path]: Path to read from or write to, supports glob-pattern matching when reading

Optional arguments:
    [PROJECT]: Google Cloud Project ID, If not passed, falls back to the default inferred from the environment.
"""
```